File: tests/resources/metrics_sample.rs
// Sample Rust file used to check the metrics.
use std::collections::HashMap;

/* A block comment
   spanning lines. */
/// A doc comment describing the function.
fn classify(values: &[i32]) -> i32 {
    let mut score = 0;
    let formatter = format!("{}", score);
    let identifier = formatter.len() as i32;
    let mut returned = HashMap::new();
    returned.insert("iffy", identifier);
    for value in values {
        if *value > 10 {
            score += identifier;
        } else if *value < 0 {
            continue;
        } else {
            score -= 1;
        }
    }
    match score {
        1 | 2 => score += 10,
        x if x > 100 => break_even(&mut score),
        _ => score += 3,
    }
    let mut i = 0;
    loop {
        i += 1;
        if i > 3 {
            break;
        }
    }
    while let Some(value) = values.iter().next() {
        if *value == 0 {
            break;
        }
        i -= 1;
    }
    while i > 0 {
        i -= 1;
    }
    assert!(score != 0);
    return score;
}

fn break_even(score: &mut i32) {
    *score = 0;
}

File: tests/resources/metrics_sample.dart
// Sample Dart file used to check the metrics.
import 'dart:math';

/* A block comment
   spanning lines. */
/// A doc comment describing the class.
class Classifier {
  final elsewhere = 'iffy';
  final returned = <String, int>{};

  int doubled(int value) => value * 2;

  int classify(List<int> values) {
    var score = 0;
    for (final value in values) {
      if(value > 10) {
        score += elsewhere.length;
      } else if (value < 0) {
        continue;
      } else {
        score -= 1;
      }
    }
    switch (elsewhere) {
      case 'high':
        score += 10;
        break;
      case 'low' || 'none':
        score += 5;
      default:
        score += 1;
    }
    final label = switch (score) {
      0 => 'none',
      _ => 'some',
    };
    var i = 0;
    while(i < 3) {
      i++;
    }
    try {
      returned['breakEven'] = score;
    } on FormatException catch (e) {
      assert(score != 0);
    }
    return score + label.length;
  }
}
